core/src/test/java/org/elasticsearch/aliases/IndexAliasesIT.java

```diff
@@ -785,6 +803,21 @@ public void testCreateIndexWithAliasesFilterNotValid() {
         }
     }
 
+    // aliases can be added only to indices
+    public void testAddAliasesOnlyToIndices() throws Exception {
+        logger.info("--> creating index [2017-05-20]");
+        assertAcked(prepareCreate("2017-05-20"));
+        ensureGreen();
+
+        logger.info("--> adding [week_20] alias to [2017-05-20]");
+        assertAcked(admin().indices().prepareAliases().addAlias("2017-05-20", "week_20"));
+
+        expectThrows(IndexNotFoundException.class, () -> admin().indices().prepareAliases()
+                .addAliasAction(AliasActions.add().index("week_20").alias("tmp")).execute().actionGet());
+
+        assertAcked(admin().indices().prepareAliases().addAliasAction(AliasActions.add().index("2017-05-20").alias("tmp")).execute().get());
+    }
+
```

Comment:
`IndexAliasesIT#testRemoveIndexAndReplaceWithAlias` kind of tests this. I could add a test which explicitly tests this though.